hpat/hiframes/aggregate.py

```diff
@@ -1047,51 +1047,54 @@ def gen_top_level_agg_func(key_names, return_key, red_var_typs, out_typs,
 
 
 def compile_to_optimized_ir(func, arg_typs, typingctx):
+    state = namedtuple('State',
+                            ['typingctx', 'targetctx', 'args', 'func_ir', 'typemap', 'return_type',
+                                'calltypes', 'metadata'])
```

Comment:
Just in case there is another way to define named tuple:
```
from typing import NamedTuple

class State(NamedTuple):
    typingctx: <type>
    targetctx: <type>
    args: <type>
    func_ir: <type>
    typemap: <type>
    return_type: <type>
    calltypes: <type>
    metadata: <type>
```

Reply:
If you don't have strong opinion it should be modified, I'd prefer to left as is